enhanceWithPromiseSupport.js

```diff
@@ -0,0 +1,12 @@
+// if a callback is provided then don't resolve but use provided callback
+export const enhanceWithPromiseSupport = notification => ({
```

Comment:
Can you share the reason why this is a function? I would have thought we could just update `FlowAlert` itself to always support an optional promise? We're not relying on `confirm` etc.'s return value are we?

Reply:
At first I did exactly that but realized that I basically had the same code in `cordova.js` and `browser.js`. I thought that this would be an easier way of sharing the code, without making it dependent on any of them.

Reply:
Ok cool, can you add a comment explaining this?